crate/tee_attestation/src/lib.rs

```diff
@@ -149,23 +149,42 @@ pub fn forge_report_data_with_nonce(nonce: &[u8; 32], data: &[u8]) -> Result<Vec
 
 /// Generate a quote for the current tee
 #[cfg(target_os = "linux")]
-pub fn get_quote(report_data: &[u8]) -> Result<Vec<u8>, Error> {
-    match guess_tee()? {
-        TeeType::Sev => Ok(sev_quote::quote::get_quote(
-            &pad_report_data(report_data, sev_quote::REPORT_DATA_SIZE)?
-                .try_into()
-                .map_err(|_| Error::InvalidFormat("Report data malformed".to_owned()))?,
-        )?),
-        TeeType::Sgx => Ok(sgx_quote::quote::get_quote(
-            &pad_report_data(report_data, sgx_quote::REPORT_DATA_SIZE)?
-                .try_into()
-                .map_err(|_| Error::InvalidFormat("Report data malformed".to_owned()))?,
-        )?),
-        TeeType::Tdx => Ok(tdx_quote::quote::get_quote(
-            &pad_report_data(report_data, tdx_quote::REPORT_DATA_SIZE)?
-                .try_into()
-                .map_err(|_| Error::InvalidFormat("Report data malformed".to_owned()))?,
-        )?),
+pub fn get_quote(report_data: Option<&[u8]>) -> Result<Vec<u8>, Error> {
+    let report_data = if let Some(data) = report_data {
+        pad_report_data(data, 64)?
```

Comment:
create a const variable to store this 64 value?

Reply:
Fixed in fd24eca58d5fd65c769ac190b47c7aab26828d4b.